.github/workflows/build_wheels_linux.yml

```diff
@@ -0,0 +1,47 @@
+name: Build Linux Wheels
+
+on:
+  pull_request:
+  push:
+    branches:
+      - nightly
+  workflow_dispatch:
+
+jobs:
+  generate-matrix:
+    uses: pytorch/test-infra/.github/workflows/generate_binary_build_matrix.yml@main
+    with:
+      package-type: wheel
+      os: linux
+      test-infra-repository: pytorch/test-infra
+      test-infra-ref: main
+      with-cuda: disable
+  build:
+    needs: generate-matrix
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - repository: pytorch/data
+            pre-script: ""
+            post-script: ""
+            smoke-test-script: test/smoke_test/smoke_test.py
+            package-name: torchdata
+    name: ${{ matrix.repository }}
+    uses: pytorch/test-infra/.github/workflows/build_wheels_linux.yml@main
+    with:
+      repository: ${{ matrix.repository }}
+      ref: ""
+      test-infra-repository: pytorch/test-infra
+      test-infra-ref: main
+      build-matrix: ${{ needs.generate-matrix.outputs.matrix }}
+      pre-script: ${{ matrix.pre-script }}
```

Comment:
I think we need to do an extra step in `pre-script` for linux wheels. This is needed to convert torchdata wheels to `manylinux` compatible. I just checked the 3.7 release, it's not `manylinux2014` compatible by `auditwheel show torchdata-0.6.0.dev20221216+cpu-cp37-cp37m-linux_x86_64.whl`. And, apparently, `auditwheel repair` won't help.

Reason behind it is our C-Extension relying on CMake system from AWSSDK, which requires OpenSSL. I am not sure my approach is the best to make it manylinux compatible. Just want to share what I have done, I basically install static OpenSSL/libcrypto library manually.
https://github.com/pytorch/data/blob/0a0ae5d35be439c786c9ab798ea1596238d8640a/.github/workflows/_build_test_upload.yml#L124-L131 and specify those paths to let AWSSDK statically linking against OpenSSL https://github.com/pytorch/data/blob/0a0ae5d35be439c786c9ab798ea1596238d8640a/.github/workflows/_build_test_upload.yml#L153-L156

Reply:
It should be similar to what TorchAudio is doing in https://github.com/pytorch/audio/blob/main/packaging/pre_build_script.sh

Reply:
We have the `pre-script` option exactly for this purpose. I agree, let's merge this PR and then start working on follow up PR to support manylinux compatible.